geopandas/tests/test_geoseries.py

```diff
@@ -538,13 +538,40 @@ def test_from_series(self):
         assert s.index is g.index
 
     # GH 1216
-    def test_expanddim(self):
+    @pytest.mark.parametrize("name", [None, "geometry", "Points"])
+    @pytest.mark.parametrize("crs", [None, "epsg:4326"])
+    def test_reset_index(self, name, crs):
         s = GeoSeries(
-            [MultiPoint([(0, 0), (1, 1)]), MultiPoint([(2, 2), (3, 3), (4, 4)])]
+            [MultiPoint([(0, 0), (1, 1)]), MultiPoint([(2, 2), (3, 3), (4, 4)])],
+            name=name,
+            crs=crs,
         )
         s = s.explode(index_parts=True)
         df = s.reset_index()
         assert type(df) == GeoDataFrame
+        # name None -> 0, otherwise name preserved
+        assert df.geometry.name == (name if name is not None else 0)
+        assert df.crs == s.crs
+
+    @pytest.mark.parametrize("name", [None, "geometry", "Points"])
+    @pytest.mark.parametrize("crs", [None, "epsg:4326"])
+    def test_to_frame(self, name, crs):
+        s = GeoSeries(
+            [MultiPoint([(0, 0), (1, 1)]), MultiPoint([(2, 2), (3, 3), (4, 4)])],
+            name=name,
+            crs=crs,
+        )
+        df = s.to_frame()
+        assert type(df) == GeoDataFrame
+        # name None -> 0, otherwise name preserved
+        assert df.geometry.name == (name if name is not None else 0)
```

Comment:
Should we also check `df._geometry_column_name` here?

Reply:
Although that is probably also tested in `test_to_frame` to test_op_output_types.py

Reply:
Yes we should. Until you pointed this out I hadn't realised that if `df.geometry.name` wasn't an exception that it could be different to `df._geometry_column_name`. But that's totally wrong (it's never occured to me before to give a series in a dataframe a name other than just renaming the column)
```python
cities = geopandas.read_file(geopandas.datasets.get_path('naturalearth_cities'))
In [24]: cities.geometry.name = 'foo'

In [25]: cities.head(2)
Out[25]:
           name                   geometry
0  Vatican City  POINT (12.45339 41.90328)
1    San Marino  POINT (12.44177 43.93610)

In [26]: cities.geometry.name
Out[26]: 'foo'

In [27]: cities._geometry_column_name
Out[27]: 'geometry'
```

Reply:
Ah, I wasn't actually even thinking about that :) 

I was thinking that in theory you could have a mismatch between the actual name of the geometry column in the GeoDataFrame, and what we store in `_geometry_column_name` (eg the actual dataframe could have a "geom" column, but the `_geometry_column_name` thinks it is "geometry", so those got out of sync). But, since we use `_geometry_column_name` to look up the column when doing `df.geometry`, this is in practice not possible to be out of sync? (or at least, when it _is_ out of sync, doing `df.geometry` would raise an error, and the test as it as would already fail).

Now, what you point out is actually a generic issue with "caching" of the returned Series for a column, which is something that pandas does. A non-geopandas example:

```
In [12]: df = pd.DataFrame({'a': [1, 2, 3]})

In [13]: df
Out[13]: 
   a
0  1
1  2
2  3

In [14]: df["a"]
Out[14]: 
0    1
1    2
2    3
Name: a, dtype: int64

In [15]: df["a"].name = "blabla"

In [16]: df
Out[16]: 
   a
0  1
1  2
2  3

In [17]: df["a"]
Out[17]: 
0    1
1    2
2    3
Name: blabla, dtype: int64   # <--- wrong name !!
```

Personally, I would consider this a bug in pandas that you can mutate the cached return value that way. But in any case I don't think we need to consider that corner case here. Users shouldn't do ``cities.geometry.name = 'foo'`` and expect this to update `cities` :)

Reply:
Agree with everything you've said above
>this is in practice not possible to be out of sync? (or at least, when it is out of sync, doing df.geometry would raise an error, and the test as it as would already fail).

This is what I was originally going to reply with, that they have to be consistent, and if they're not then `df.geometry.name` is an attribute error before I went down the corner case rabbit hole - which I don't think we should do anything about.

(But I've now put that check in and there's no real harm in checking both, I will leave it)